File: 2018/01-01.js
const { readFileSync } = require('fs')

const input = readFileSync('./01-01.input.txt', 'utf-8')

const result = input
  .trim()
  .split('\n')
  .map(v => parseInt(v, 10))
  .reduce((acc, v) => acc + v, 0)

console.log({ result })

File: 2018/01-01.hs
module Main where

main :: IO ()
main = do
  s <- readFile "01-01.input.txt"
  print $ foldl f 0 $ lines s

f :: Integer -> String -> Integer
f acc ('+':n) = acc + (read n :: Integer)
f acc ('-':n) = acc - (read n :: Integer)
f acc _ = acc
